foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/version_specific/SparkBomDependencyMigration.java

```diff
@@ -0,0 +1,86 @@
+package com.boozallen.aissemble.upgrade.migration.version_specific;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractPomMigration;
+import org.apache.maven.model.Model;
+import org.apache.maven.project.MavenProject;
+import org.technologybrewery.baton.util.pom.PomHelper;
+import org.technologybrewery.baton.util.pom.PomModifications;
+
+import java.io.File;
+
+import static org.apache.commons.lang3.StringUtils.repeat;
+import static org.technologybrewery.baton.util.pom.LocationAwareMavenReader.END;
+
+public class SparkBomDependencyMigration extends AbstractPomMigration {
+
+    @Override
+    protected boolean shouldExecuteOnFile(File pomFile) {
+        Model model = PomHelper.getLocationAnnotatedModel(pomFile);
```

Comment:
a: Dont think you need the locations here
```suggestion
        Model model = getMavenProject().getModel();
```

Reply:
Got rid of location annotated model here

Reply:
Actually on a second look at this, changing to `getMavenProject().getModel();` was causing issues on a test downstream project where it couldn't correctly determine that the bom already existed and kept adding it each time the migration ran. reverting to `PomHelper.getLocationAnnotatedModel(pomFile);` solved this issue.

Reply:
Would be interesting, but not super critical, to understand _why_ `getMavenProject().getModel` doesn't work.  One difference between the two is that `PomHelper` will parse the pom file as it currently exists, whereas `getMavenProject().getModel()` is the model as it existed when the Maven execution _began_. However, that doesn't necessarily explain why it causes the issue described above.